opba-auth-rest-impl/src/main/java/de/adorsys/opba/tppauthapi/controller/PsuAuthController.java

```diff
@@ -0,0 +1,114 @@
+package de.adorsys.opba.tppauthapi.controller;
+
+import com.nimbusds.jose.JWSAlgorithm;
+import com.nimbusds.jose.JWSHeader;
+import com.nimbusds.jose.crypto.RSASSASigner;
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.SignedJWT;
+import de.adorsys.opba.db.domain.entity.psu.Psu;
+import de.adorsys.opba.tppauthapi.config.CookieProperties;
+import de.adorsys.opba.tppauthapi.model.generated.PsuAuthBody;
+import de.adorsys.opba.tppauthapi.resource.generated.PsuAuthApi;
+import de.adorsys.opba.tppauthapi.service.PsuAuthService;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.bouncycastle.util.encoders.Hex;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseCookie;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+
+import java.nio.charset.StandardCharsets;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.time.Duration;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.Date;
+import java.util.Optional;
+import java.util.UUID;
+
+import static de.adorsys.opba.restapi.shared.HttpHeaders.AUTHORIZATION_SESSION_ID;
+import static de.adorsys.opba.restapi.shared.HttpHeaders.X_REQUEST_ID;
+import static de.adorsys.opba.restapi.shared.HttpHeaders.X_XSRF_TOKEN;
+import static org.springframework.http.HttpHeaders.SET_COOKIE;
+
+@Slf4j
+@Controller
+@RequiredArgsConstructor
+public class PsuAuthController implements PsuAuthApi {
+    public static final Duration WEEK = Duration.ofDays(7);
+
+    private final PsuAuthService psuAuthService;
+    private final CookieProperties cookieProperties;
+
+    @Override
+    @SneakyThrows
+    public ResponseEntity<String> login(PsuAuthBody psuAuthBody, UUID xRequestID) {
+        Optional<Psu> psu = psuAuthService.getPsu(psuAuthBody.getId());
+        if (!psu.isPresent()) {
+            return new ResponseEntity<>(HttpStatus.UNAUTHORIZED);
+        }
+
+        String jwtToken = generateToken(psu.get().getUserId(), getPrivateKey());
+
+        String sessionCookieString = ResponseCookie.from(AUTHORIZATION_SESSION_ID, jwtToken)
```

Comment:
Instead of this, you can provide bean ResponseCookie.Builder and use it. Also you can use Configuration properties on ResponseCookie.Builder too.

Reply:
done